### PR TITLE
Catch VM Service disappearance from run/attach handler code

### LINKED ISSUE #75297: Handle Service disappeared RPC error in run/attach outer layers
Rather than crashing the tool of course

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/75297

Catch kServiceDisappeared from run/attach handler code

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/run_test.dart [hunk 1]
```diff
@@ -542,6 +547,30 @@ void main() {
       expect(dartDefines, <String>['FLUTTER_WEB_AUTO_DETECT=false','FLUTTER_WEB_USE_SKIA=false']);
     });
   });
+
+  testUsingContext('Flutter run catches service has disappear errors and throws a tool exit', () async {
+    final FakeResidentRunner residentRunner = FakeResidentRunner();
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/hermetic/run_test.dart:552]: How awful would it be to get this to boil down to a fake vm service host that throws the rpc error?

**Review by zanderso — APPROVED** [review]
lgtm w/ optional suggestion for a test that might more accurately simulate the bug.

File: packages/flutter_tools/test/commands.shard/hermetic/run_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/hermetic/run_test.dart:552]: The resident runner does a tremendous amount of setup. It would be bad

**Comment by jonahwilliams** [comment]
g3fix applied, waiting for tree

**Pull request merged**